modules/dgi_actions_handle/src/Commands/Update.php

```diff
@@ -0,0 +1,196 @@
+<?php
+
+namespace Drupal\dgi_actions_handle\Commands;
+
+use Drupal\dgi_actions\Entity\IdentifierInterface;
+use Drupal\dgi_actions_handle\Utility\HandleTrait;
+use GuzzleHttp\ClientInterface;
+use GuzzleHttp\Exception\RequestException;
+
+/**
+ * Separate class that handles Handle updates for convenience.
+ */
+class Update {
+
+  use HandleTrait;
+
+  /**
+   * The type of request that's going to be executed; either "GET" or "PUT".
+   *
+   * @var string
+   */
+  protected $requestType;
+
+  /**
+   * The identifier to be used.
+   *
+   * @var \Drupal\dgi_actions\Entity\IdentifierInterface
+   */
+  protected $identifier;
+
+  /**
+   * The HTTP client to be used.
+   *
+   * @var \GuzzleHttp\ClientInterface
+   */
+  protected $client;
+
+  /**
+   * The handle to be updated.
+   *
+   * @var string
+   */
+  protected string $handle;
+
+  /**
+   * The location to be updated.
+   *
+   * @var string
+   */
+  protected string $targetLocation;
+
+  /**
+   * The index to be updated.
+   *
+   * @var null|string
+   */
+  protected ?string $indexToUpdate;
+
+  /**
+   * Handles updating a handle via Drush.
+   *
+   * @param \Drupal\dgi_actions\Entity\IdentifierInterface $identifier
+   *   The identifier to be used.
+   * @param \GuzzleHttp\ClientInterface $client
+   *   The HTTP client to be used.
+   * @param array $options
+   *   An array containing:
+   *   -handle (string): The handle to be updated.
+   *   -target_location (string): The URL to be updated to.
+   */
+  public function __construct(IdentifierInterface $identifier, ClientInterface $client, array $options) {
+    $this->identifier = $identifier;
+    $this->client = $client;
+    $this->handle = $options['handle'];
+    $this->targetLocation = $options['target_location'];
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getIdentifier(): IdentifierInterface {
+    return $this->identifier;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function getUri(): string {
+    return "{$this->getIdentifier()->getServiceData()->getData()['host']}/{$this->handle}";
+  }
+
+  /**
+   * Updates a handle.
+   */
+  public function updateHandle(): void {
+    try {
+      $get_handle = $this->retrieveHandle();
+    }
+    catch (RequestException $e) {
+      throw new \Exception(dt('Failed to retrieve the Handle. Error: !message.', [
+        '!message' => $e->getMessage(),
+      ]));
+    }
+
+    // Find the index that needs to be updated.
+    foreach ($get_handle['values'] as $values) {
```

Comment:
Is it possible for multiple values to require updating? This seems like it would only update the first one found.

Reply:
Possibly? This is what we did in the 7 update version. Handle indices and their intended use seem to vary depending on the implementer.

Reply:
Good enough for i7 good enough for me.